# Etat des lieux initial du dataset : btc_macro_dataset.csv

In [1]:
import sys

print(sys.executable)

d:\ProjectFolderDevAI_2025-2026\Certification BLOCS 2&3\Projet-VolatiChainXplorerAI\.venv\Scripts\python.exe


In [10]:
from pathlib import Path

root_projet = Path("__file__").resolve().parents[1]
data_path = root_projet / "data" / "exports" / "datasets" / "btc_macro_dataset.csv"
print("Dossier courant :", Path.cwd())
print(data_path)
print("Fichier trouvé :", data_path.exists())

Dossier courant : d:\ProjectFolderDevAI_2025-2026\Certification BLOCS 2&3\Projet-VolatiChainXplorerAI\notebooks
D:\ProjectFolderDevAI_2025-2026\Certification BLOCS 2&3\Projet-VolatiChainXplorerAI\data\exports\datasets\btc_macro_dataset.csv
Fichier trouvé : True


In [3]:
import pandas as pd

df = pd.read_csv(data_path)

df.head(10)

,date,open,high,low,close,volume,market_cap,rate_mro,inflation_rate,unemployment_rate,monetary_m3_rate
0,2010-07-14,0.045655,0.048348,0.038189,0.044277,205.314915,149358.200915,1.0,1.7,10.1,0.087906
1,2010-07-15,0.043730,0.052686,0.041843,0.044633,345.633197,151946.912726,1.0,1.7,10.1,0.087906
2,2010-07-16,0.044836,0.055827,0.044436,0.051399,384.382207,176284.305020,1.0,1.7,10.1,0.087906
3,2010-07-17,0.051405,0.059957,0.044286,0.050900,15.452590,175401.041475,1.0,1.7,10.1,0.087906
4,2010-07-18,0.050974,0.062366,0.049533,0.060789,57.947844,209538.725751,1.0,1.7,10.1,0.087906
5,2010-07-19,0.060778,0.064544,0.053298,0.057261,442.140585,197686.305250,1.0,1.7,10.1,0.087906
6,2010-07-20,0.057533,0.060156,0.049891,0.055843,203.597504,193480.201829,1.0,1.7,10.1,0.087906
7,2010-07-21,0.056245,0.059784,0.050237,0.058165,447.871547,202127.296414,1.0,1.7,10.1,0.087906
8,2010-07-22,0.057558,0.058218,0.039598,0.045683,1681.489031,159153.185842,1.0,1.7,10.1,0.087906
9,2010-07-23,0.045687,0.053430,0.043203,0.048576,1861.039464,169794.272526,1.0,1.7,10.1,0.087906


In [4]:
print("Dimensions :", df.shape)
print("\nColonnes :", df.columns.tolist())
print("\nTypes :\n", df.dtypes)
print("\nAperçu :\n", df.head())
print("\nValeurs manquantes :\n", df.isna().sum())
print("\nNombre de doublons :", df.duplicated().sum())
print("\nStatistiques :\n", df.describe())

Dimensions : (5903, 11)

Colonnes : ['date', 'open', 'high', 'low', 'close', 'volume', 'market_cap', 'rate_mro', 'inflation_rate', 'unemployment_rate', 'monetary_m3_rate']

Types :
 date                  object
open                 float64
high                 float64
low                  float64
close                float64
volume               float64
market_cap           float64
rate_mro             float64
inflation_rate       float64
unemployment_rate    float64
monetary_m3_rate     float64
dtype: object

Aperçu :
          date      open      high       low     close      volume  \
0  2010-07-14  0.045655  0.048348  0.038189  0.044277  205.314915   
1  2010-07-15  0.043730  0.052686  0.041843  0.044633  345.633197   
2  2010-07-16  0.044836  0.055827  0.044436  0.051399  384.382207   
3  2010-07-17  0.051405  0.059957  0.044286  0.050900   15.452590   
4  2010-07-18  0.050974  0.062366  0.049533  0.060789   57.947844   

      market_cap  rate_mro  inflation_rate  unemployment_ra

In [5]:
# Conversion de la date
df["date"] = pd.to_datetime(df["date"], errors="coerce")

# Contrôles temporels
print("Dates invalides :", df["date"].isna().sum())
print("Première date :", df["date"].min())
print("Dernière date :", df["date"].max())
print("Dates triées :", df["date"].is_monotonic_increasing)
print("Dates dupliquées :", df["date"].duplicated().sum())

# Recherche des jours absents
dates_attendues = pd.date_range(
    start=df["date"].min(),
    end=df["date"].max(),
    freq="D"
)

dates_manquantes = dates_attendues.difference(df["date"])

print("Nombre de dates manquantes :", len(dates_manquantes))
print("Premières dates manquantes :", dates_manquantes[:10])

# Analyse des valeurs macroéconomiques manquantes
macro_columns = [
    "rate_mro",
    "inflation_rate",
    "unemployment_rate",
    "monetary_m3_rate"
]

lignes_macro_manquantes = df[df[macro_columns].isna().any(axis=1)]

print("Première date macro manquante :", lignes_macro_manquantes["date"].min())
print("Dernière date macro manquante :", lignes_macro_manquantes["date"].max())
display(lignes_macro_manquantes.head())
display(lignes_macro_manquantes.tail())

# Contrôle du volume
print("Volumes égaux à zéro :", (df["volume"] == 0).sum())
display(df.loc[df["volume"] == 0, ["date", "volume"]].head(10))

Dates invalides : 0
Première date : 2010-07-14 00:00:00
Dernière date : 2026-09-12 00:00:00
Dates triées : True
Dates dupliquées : 0
Nombre de dates manquantes : 2
Premières dates manquantes : DatetimeIndex(['2022-06-28', '2022-06-29'], dtype='datetime64[ns]', freq='D')
Première date macro manquante : NaT
Dernière date macro manquante : NaT


,date,open,high,low,close,volume,market_cap,rate_mro,inflation_rate,unemployment_rate,monetary_m3_rate


,date,open,high,low,close,volume,market_cap,rate_mro,inflation_rate,unemployment_rate,monetary_m3_rate


Volumes égaux à zéro : 248


,date,volume
341,2011-06-20,0.0
342,2011-06-21,0.0
343,2011-06-22,0.0
344,2011-06-23,0.0
345,2011-06-24,0.0
346,2011-06-25,0.0
1020,2013-04-29,0.0
1021,2013-04-30,0.0
1022,2013-05-01,0.0
1023,2013-05-02,0.0


In [6]:
# Observer les données autour des deux dates absentes
display(
    df.loc[
        df["date"].between("2022-06-25", "2022-07-02"),
        ["date", "open", "high", "low", "close", "volume", "market_cap"]
    ]
)

# Nombre de volumes nuls par année
volume_zero = df.loc[df["volume"] == 0, ["date", "volume"]].copy()
volume_zero["annee"] = volume_zero["date"].dt.year

display(
    volume_zero.groupby("annee")
    .size()
    .rename("nombre_volume_zero")
)

# Identifier les périodes consécutives de volume nul
volume_zero["groupe"] = (
    volume_zero["date"].diff().dt.days.ne(1).cumsum()
)

periodes_volume_zero = (
    volume_zero.groupby("groupe")
    .agg(
        debut=("date", "min"),
        fin=("date", "max"),
        nombre_jours=("date", "size")
    )
)

display(periodes_volume_zero)

# Vérifier la cohérence des prix OHLC
ohlc_invalides = df[
    (df["high"] < df[["open", "close", "low"]].max(axis=1))
    | (df["low"] > df[["open", "close", "high"]].min(axis=1))
    | (df[["open", "high", "low", "close"]] <= 0).any(axis=1)
]

print("Lignes OHLC incohérentes :", len(ohlc_invalides))

,date,open,high,low,close,volume,market_cap
4364,2022-06-25,20115.214406,20387.386543,19860.359351,20369.789007,1.740484e+10,3.885965e+11
4365,2022-06-26,20364.252865,20636.354199,19896.195040,19896.195040,1.707766e+10,3.800286e+11
4366,2022-06-27,19902.176070,20292.885437,19475.282605,19596.063434,1.984298e+10,3.744221e+11
4367,2022-06-30,19249.968917,19289.894218,17866.481291,18881.949425,2.514600e+10,3.614111e+11
4368,2022-07-01,18915.447972,19699.641234,18237.192775,18476.547863,2.936267e+10,3.509194e+11
4369,2022-07-02,18481.792419,18574.716472,18244.231120,18450.552217,1.735570e+10,3.521001e+11


annee
2011      6
2013    242
Name: nombre_volume_zero, dtype: int64

,debut,fin,nombre_jours
groupe,,,
1,2011-06-20,2011-06-25,6
2,2013-04-29,2013-12-26,242


Lignes OHLC incohérentes : 0


In [7]:
print("Volumes négatifs :", (df["volume"] < 0).sum())
print(
    "Capitalisations nulles ou négatives :",
    (df["market_cap"] <= 0).sum()
)

Volumes négatifs : 0
Capitalisations nulles ou négatives : 0


## Conclusion de l’état des lieux initial

Le dataset contient 5 802 observations et 11 variables, du 14 juillet 2010 au 3 juin 2026.

### Contrôles validés

- Dates invalides : 0
- Dates dupliquées : 0
- Ordre chronologique : valide
- Lignes OHLC incohérentes : 0
- Volumes négatifs : 0
- Capitalisations nulles ou négatives : 0

### Anomalies détectées

- Deux dates manquantes : 28 et 29 juin 2022
- 34 lignes macroéconomiques manquantes, du 1er mai au 3 juin 2026
- 248 volumes nuls :
  - 6 jours en juin 2011
  - 242 jours entre avril et décembre 2013

### Décision

Les anomalies ne seront pas corrigées arbitrairement. Une nouvelle collecte sera d’abord exécutée afin de récupérer les données disponibles auprès des sources.

In [11]:
import pandas as pd

df = pd.read_csv(
    "../data/exports/datasets/btc_macro_dataset.csv",
    parse_dates=["date"]
)

expected_dates = pd.date_range(
    start=df["date"].min(),
    end=df["date"].max(),
    freq="D"
)

missing_dates = expected_dates.difference(df["date"])

print("Nombre de dates manquantes :", len(missing_dates))
print("Dates manquantes :", missing_dates.tolist())

Nombre de dates manquantes : 2
Dates manquantes : [Timestamp('2022-06-28 00:00:00'), Timestamp('2022-06-29 00:00:00')]


In [14]:
df.loc[
    df["date"].between("2022-06-25", "2022-07-02")
]

,date,open,high,low,close,volume,market_cap,rate_mro,inflation_rate,unemployment_rate,monetary_m3_rate
4364,2022-06-25,20115.214406,20387.386543,19860.359351,20369.789007,1.740484e+10,3.885965e+11,0.0,8.6,6.8,5.966328
4365,2022-06-26,20364.252865,20636.354199,19896.195040,19896.195040,1.707766e+10,3.800286e+11,0.0,8.6,6.8,5.966328
4366,2022-06-27,19902.176070,20292.885437,19475.282605,19596.063434,1.984298e+10,3.744221e+11,0.0,8.6,6.8,5.966328
4367,2022-06-30,19249.968917,19289.894218,17866.481291,18881.949425,2.514600e+10,3.614111e+11,0.0,8.6,6.8,5.966328
4368,2022-07-01,18915.447972,19699.641234,18237.192775,18476.547863,2.936267e+10,3.509194e+11,0.0,8.9,6.7,5.770509
4369,2022-07-02,18481.792419,18574.716472,18244.231120,18450.552217,1.735570e+10,3.521001e+11,0.0,8.9,6.7,5.770509
